runtime/src/main/java/com/dropbox/kaiken/runtime/KaikenRuntimeTestUtils.kt

```diff
@@ -0,0 +1,104 @@
+package com.dropbox.kaiken.runtime
+
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelStore
+import com.dropbox.kaiken.Injector
+import com.dropbox.kaiken.annotation.InternalKaikenApi
+import com.dropbox.kaiken.runtime.KaikenRuntimeTestUtils.testMode
+import kotlin.reflect.KClass
+
+object KaikenRuntimeTestUtils {
+    private val injectorFactoryOverrides = mutableMapOf<KClass<out InjectorHolder<*>>, InjectorFactory<*>>()
+    private val injectorOverrides = mutableMapOf<KClass<out Any>, InjectorHolder<Injector>>()
+
+    @InternalKaikenApi
+    var testMode: Boolean = false
```

Comment:
Nit fakeMode rather than test mode as we may use this in debugging as well. Up to you

Reply:
I think I'd like to keep as `test` for now until we have a good debugging use case. Because right now this completely changes some behavior to make it incompatible with production, as to make it incompatible with debugging.

Reply:
Cool. My use case is a sample/sandbox app in a large code base. I'd like to override injectors for implementation  that do not exist on path.

We should also think about doing what mavericks does and short circuiting if app is not debuggable (release)

Reply:
I see. I think if you need this override for sample/sandboxing in a large code base then "you're holding it wrong" :) I think it'd be better to have a debug "SandboxActivity" that contained basic user services. 

Very good point. I'd added `if (BuildConfig.Debug)` to all code points where we're accessing the override. This should also help proguard get rid of the test objets in release builds.